DEVELOPER_GUIDE.md

```diff
@@ -93,8 +97,16 @@ snapshots/
                 ├── maven-metadata.xml.sha1
                 ├── maven-metadata.xml.sha256
                 └── maven-metadata.xml.sha512
-
-
 ```
+
 1. Change the url from ``"https://aws.oss.sonatype.org/content/repositories/snapshots"`` to your local path and comment out the credentials under publishing/repositories in build.gradle.
 2. Run ```./gradlew publishPluginZipPublicationToSnapshotsRepository```.
+
+### Adding Workflow Steps
+
+To add functionality to workflows, add new Workflow Steps to the [`org.opensearch.flowframework.workflow`](https://github.com/opensearch-project/flow-framework/tree/main/src/main/java/org/opensearch/flowframework/workflow) package.
+1. Implement the [Workflow](https://github.com/opensearch-project/flow-framework/blob/main/src/main/java/org/opensearch/flowframework/workflow/WorkflowStep.java) interface. See existing steps for examples for input, output, and API execution.
+2. Choose a unique name for the step which is not used by other steps.
+3. Add a constructor and call it from the [WorkflowStepFactory](https://github.com/opensearch-project/flow-framework/blob/main/src/main/java/org/opensearch/flowframework/workflow/WorkflowStepFactory.java).
```

Comment:
Should we add a step regarding the resources section if their steps creates a resource?

Reply:
Oops! Yes, I had that thought and then that thought went away...